api/include/opentelemetry/trace/trace_state.h

```diff
@@ -45,51 +45,154 @@ class TraceState
   class Entry
   {
   public:
-    Entry() noexcept = default;
-
-    // Copy constructor.
-    Entry(const Entry &copy);
-
-    // Creates an Entry for a given key-value pair.
-    Entry(nostd::string_view key, nostd::string_view value) noexcept;
-
-    nostd::string_view GetKey();
-    nostd::string_view GetValue();
+    Entry() : key_(new char[kKeyMaxSize]), value_(new char[kValueMaxSize]){};
```

Comment:
Do we need this constructor?

Reply:
Yes, it's used in the `TraceState` constructor on line 100, when we create the `Entry` array: `entries_(new Entry[kMaxKeyValuePairs])`

Reply:
I see. Could we avoid the allocation and initialize with a `nullptr` here, for `key_` and `value_`?